ssl/tls13_enc.c

```diff
@@ -702,10 +741,14 @@ int tls13_change_cipher_state(SSL_CONNECTION *s, int which)
  err:
     if ((which & SSL3_CC_EARLY) != 0) {
         /* We up-refed this so now we need to down ref */
+        if ((EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER) == 0)
+            ssl_evp_md_free(mac_md);
```

Comment:
As we have just discussed, it's correct because the `ssl_evp_md_free(mac_md);` is done only if
`(which & SSL3_CC_EARLY) != 0 `while the mentioned assignment to `mac_md` further above
is only in case `(which & SSL3_CC_EARLY) == 0`, so in a different (non-early) situation.

Turns out that I was confused by the expression `!(which & SSL3_CC_EARLY)`, in the original line 601,
which is not according to the coding style.
So please update that original guard`!(which & SSL3_CC_EARLY)` to `(which & SSL3_CC_EARLY) == 0`,
as well as, more further up, update `(which & SSL3_CC_EARLY)` to `(which & SSL3_CC_EARLY) != 0`.

Reply:
👍